test_builder/test_builder

```diff
@@ -1397,5 +1397,123 @@ function multi_level_rpc_traffic_pattern_setup() {
   }"
 }
 
+################################################################################
+# Setup the test config proto for the variable_response_multi_level_rpc traffic pattern:
+# Arguments:
+#   None
+# Inherits and mutates:
+#   RESULTS_COLLECTION_TIMEOUT
+#   params_dict (associative array)
+################################################################################
+function variable_response_multi_level_rpc_traffic_pattern_setup() {
+  local -i TEST_DURATION="${params_dict[test_duration]}"
+  params_dict[test_timeout]="$((TEST_DURATION + RESULTS_COLLECTION_TIMEOUT))"
+
+  set_parameter_default qps 10000
+  set_parameter_default root_count 4
+  set_parameter_default leaf_count 20
+  set_parameter_default exact_size "" "For convenience you can set the exact size to an integer"
+  if [[ -v params_dict[exact_size] ]]; then
+    if [[ -v params_dict[lower_size] || -v params_dict[upper_size] ]]; then
+      log_stderr "Only exact_size OR lower_size and upper_size need to be specified"
+      exit 14
+    fi
+    #set default to print on test_builder manual
+    set_parameter_default lower_size 1024
+    set_parameter_default upper_size 64000
+    local LOWER="${params_dict[exact_size]}"
+    local UPPER="${params_dict[exact_size]}"
+  else
+    set_parameter_default lower_size 1024
+    set_parameter_default upper_size 64000
+    local LOWER="${params_dict[lower_size]}"
```

Comment:
LOWER/UPPER are always set on all paths to the same expresion. just do it once unconditionally